x-pack/platform/plugins/shared/lens/public/editor_frame_service/editor_frame/config_panel/layer_panel.tsx

```diff
@@ -73,6 +89,17 @@ export function LayerPanel(props: LayerPanelProps) {
     onDropToDimension,
     setIsInlineFlyoutVisible,
     onlyAllowSwitchToSubtypes,
+    attributes,
```

Comment:
This component is already super lengthy and difficult to read with so many props and concepts mixed and here with so many props that are specific to only a part of it (ESQL editor) it makes things even more complex. Can you consider other options? For example, instead of passing all of this, passing an extracted component as a prop (a prop `ESQLEditor` for example) that encapsulates all the logic attached to it.

Reply:
probably it would make sense to refactor this component to make use of contexts, but i think its outside the scope of this PR

Reply:
I agree this file is lengthy and complex to read. And it's definitely outside the scope to try to refactor the existing functionalities. 

However, you've also added a lot of props and lines of code that are making the component even more tangled. What I am requesting is to extract this new functionality, whether to another component that we pass as prop (preferred solution) or to another component that uses the new props.

I'm a bit unclear on your comment about what's outside the scope of this PR: are you planning to implement the changes I suggested, or do you see this as **also** being out of scope?

Reply:
@mbondyra while not a complete solution addressing your concerns, I've isolated the ESQLEditor code into its own component which makes the changes into this file less disruptive.
Of course this is just a local patch and a follow up with better state management should be planned upon these changes.